src/actions_ls.rs

```diff
@@ -207,21 +209,17 @@ impl ActionHandler {
     }
 
     pub fn complete(&self, id: usize, params: TextDocumentPositionParams, out: &Output) {
-        let vfs: &Vfs = &self.vfs;
+        let vfs = VfsWrapper(self.vfs.clone());
         let result: Vec<CompletionItem> = panic::catch_unwind(move || {
             let pos = adjust_vscode_pos_for_racer(params.position);
             let file_path = &Path::new(params.textDocument.file_name());
 
-            let cache = core::FileCache::new();
-            let session = core::Session::from_path(&cache, file_path, file_path);
-            for (path, txt) in vfs.get_cached_files() {
-                session.cache_file_contents(&path, txt);
-            }
-
-            let src = session.load_file(file_path);
+            let cache = racer::FileCache::new(vfs);
+            let session = racer::Session::new(&cache);
 
-            let pos = session.load_file(file_path).coords_to_point(pos.line, pos.character).unwrap();
-            let results = complete_from_file(&src.code, file_path, pos, &session);
+            let coords = racer::Coordinate { line: pos.line, column: pos.character };
+            let cursor = racer::Cursor::Coords(coords);
```

Comment:
I do prefer Location to Cursor. It would be cool if the client could just skip it and offer two versions of `complete_from_file`, one taking row and column and one taking an offset. Or at least have two ctors for Location/Cursor rather than needing to make a Coords object too.